internal/protocoltest/query/serializers.go

```diff
@@ -0,0 +1,1672 @@
+// Code generated by smithy-go-codegen DO NOT EDIT.
+
+package query
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws/protocol/query"
+	"github.com/aws/aws-sdk-go-v2/internal/protocoltest/query/types"
+	smithy "github.com/awslabs/smithy-go"
+	"github.com/awslabs/smithy-go/httpbinding"
+	"github.com/awslabs/smithy-go/middleware"
+	smithytime "github.com/awslabs/smithy-go/time"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+	"sort"
+)
+
+type awsAwsquery_serializeOpEmptyInputAndEmptyOutput struct {
+}
+
+func (*awsAwsquery_serializeOpEmptyInputAndEmptyOutput) ID() string {
+	return "OperationSerializer"
+}
+
+func (m *awsAwsquery_serializeOpEmptyInputAndEmptyOutput) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
+	request, ok := in.Request.(*smithyhttp.Request)
+	if !ok {
+		return out, metadata, &smithy.SerializationError{Err: fmt.Errorf("unknown transport type %T", in.Request)}
+	}
+
+	input, ok := in.Parameters.(*EmptyInputAndEmptyOutputInput)
+	if !ok {
+		return out, metadata, &smithy.SerializationError{Err: fmt.Errorf("unknown input parameters type %T", in.Parameters)}
+	}
+
+	request.Request.URL.Path = "/"
+	request.Request.Method = "POST"
+	httpBindingEncoder, err := httpbinding.NewEncoder(request.URL.Path, request.URL.RawQuery, request.Header)
+	if err != nil {
+		return out, metadata, &smithy.SerializationError{Err: err}
+	}
+	httpBindingEncoder.SetHeader("Content-Type").String("application/x-www-form-urlencoded")
```

Comment:
Minor callout, with REST-JSON v1 SDK serializer it was discovered that several of the SDKs will write a Content-Type header even though the operation has no input parameters, https://github.com/awslabs/smithy-go/issues/110 Does Query protocol have any considerations we need to handle, or is it "safe" to always send the content type

Reply:
python / cli always set it and have done for at least six years so I don't think there's any issues. The only fiddly bit is there are some sigv2 services that can have issues if you don't add the charset, but we aren't supporting sigv2.